cmd/config-current.go

```diff
@@ -324,9 +342,46 @@ func newServerConfig() *serverConfig {
 	srvCfg.Logger.HTTP = make(map[string]loggerHTTP)
 	srvCfg.Logger.HTTP["target1"] = loggerHTTP{Endpoint: "https://username:password@example.com/api"}
 
+	srvCfg.AuthValidators = authValidators{}
+	srvCfg.AccessValidators = accessValidators{}
+
 	return srvCfg
 }
 
+func updateCachedConfig(srvCfg *serverConfig) {
+	// hold the mutex lock before a new config is assigned.
+	globalServerConfigMu.Lock()
+	globalServerConfig = srvCfg
+	if !globalIsEnvCreds {
+		globalActiveCred = globalServerConfig.GetCredential()
```

Comment:
Does seem right to update these global variables, because we will be creating two sources of truths. one source of truth is globalActiveCred, globalWORMEnabled etc and the other source of truth is globalServerConfig

Reply:
This is how it is right now, I am not changing that @krishnasrinivas